spec/forms/assessor_interface/assessment_declaration_decline_form_spec.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+RSpec.describe AssessorInterface::AssessmentDeclarationDeclineForm,
+               type: :model do
+  let(:session) { {} }
+  subject(:form) do
+    described_class.new(
+      session:,
+      assessment:,
+      declaration: true,
+      recommendation_assessor_note: "Note",
+    )
+  end
+
+  context "when recommendation is started" do
+    let(:assessment) { create(:assessment, :started) }
+    it { is_expected.to validate_presence_of(:declaration) }
+  end
+
+  context "when recommendation is declined with no failure reasons" do
+    let(:assessment) { create(:assessment, :decline) }
+    it { is_expected.to validate_presence_of(:recommendation_assessor_note) }
+
+    describe "#save" do
```

Comment:
Typically you structure a spec file with the describe blocks at the top level, as all the tests relate to the same subject (the thing we’re describing) and then within each describe block there are contexts.

If you take a look at https://github.com/DFE-Digital/apply-for-qualified-teacher-status/blob/main/spec/forms/assessor_interface/age_range_subjects_form_spec.rb for example we’ve got two describe blocks, the validations and the save method, I think it would be good to follow that structure in this test so it’s consistent but also makes it easier to find expectations in the file.

Reply:
I think you're still missing a `describe "validations"` block to group together all the expectations related to validation. Ideally they'd go together so it's easier to see what's being tested in this spec file.